scripts/cnode-helper-scripts/cntools.sh

```diff
@@ -3065,6 +3067,238 @@ function main {
+              getBalance ${pay_addr}
+              pay_lovelace=${assets[lovelace]}
+              if [[ ${pay_lovelace} -gt 0 && ${base_lovelace} -gt 0 ]]; then
+                # Both payment and base address available with funds, let user choose what to use
+                println DEBUG "\n# Select wallet address to use"
+                if [[ -n ${wallet_count} && ${wallet_count} -gt ${WALLET_SELECTION_FILTER_LIMIT} ]]; then
+                  println DEBUG "$(printf "%s\t\t${FG_LBLUE}%s${NC} Ada" "Funds :"  "$(formatLovelace ${base_lovelace})")"
+                  println DEBUG "$(printf "%s\t${FG_LBLUE}%s${NC} Ada" "Enterprise Funds :"  "$(formatLovelace ${pay_lovelace})")"
+                fi
+                select_opt "[b] Base (default)" "[e] Enterprise" "[Esc] Cancel"
+                case $? in
+                  0) addr="${base_addr}" ;;
+                  1) addr="${pay_addr}" ;;
+                  2) continue ;;
+                esac
+              elif [[ ${pay_lovelace} -gt 0 ]]; then
+                addr="${pay_addr}"
+                if [[ -n ${wallet_count} && ${wallet_count} -gt ${WALLET_SELECTION_FILTER_LIMIT} ]]; then
+                  println DEBUG "\n$(printf "%s\t${FG_LBLUE}%s${NC} Ada" "Enterprise Funds :"  "$(formatLovelace ${pay_lovelace})")"
+                fi
+              elif [[ ${base_lovelace} -gt 0 ]]; then
+                addr="${base_addr}"
+                if [[ -n ${wallet_count} && ${wallet_count} -gt ${WALLET_SELECTION_FILTER_LIMIT} ]]; then
+                  println DEBUG "\n$(printf "%s\t\t${FG_LBLUE}%s${NC} Ada" "Funds :"  "$(formatLovelace ${base_lovelace})")"
+                fi
+              else
+                println ERROR "\n${FG_RED}ERROR${NC}: no funds available for wallet ${FG_GREEN}${wallet_name}${NC}"
+                waitForInput && continue
+              fi
+              epoch=$(getEpoch)
+              echo 
+              echo "Current ${NETWORK_NAME} epoch: ${epoch}"
+              NETWORK_NAME_LOWER=$(echo "$NETWORK_NAME" | awk '{print tolower($0)}')
+              println LOG "Query ${NETWORK_NAME} polls ..."
+              println ACTION "curl -sSL -f -H \"Content-Type: application/json\" ${CIP0094_POLL_URL}"
+              ! polls=$(curl -sSL -f -H "Content-Type: application/json"  "${CIP0094_POLL_URL}" | jq -r .networks.${NETWORK_NAME_LOWER} 2>&1) && error_msg=${polls} && waitForInput && continue
+              poll_index=$(echo $polls | jq '. | length')
+              if [[ "$poll_index" -gt 0 ]]; then
+                poll_index_act=0
+                poll_index_cnt=0
+                declare poll_index_txIds=()
+                declare poll_index_titles=()
+                echo "Polls currently open for Pool answers:"
+                while read poll; do
+                  poll_index_cnt=$((poll_index_cnt+1))
+                  if [[ "$epoch" -ge "$(jq '.epoch_cast' <<< $poll)" ]] && [[ "$epoch" -lt "$(jq '.epoch_delegation' <<< $poll)" ]]; then
```

Comment:
Initial `if [[ "$poll_index" -gt 0 ]]; then` check is good but a second check is needed after filtering out invalid polls and break out as list could be empty after this.

Reply:
added this second check, and reordered the whole steps. 
This check for active polls is now the first step. only if there are active polls, it comes to the selection of pool, payment wallet and the poll answer.